webpack/assets/javascripts/react_app/redux/API/APIActions.js

```diff
@@ -0,0 +1,12 @@
+import { API_OPERATIONS } from './APIConstants';
+
+/**
+ * an API action creator for API get.
+ * @param { Object } options options for the API get acion.
+ * @param { String } options.key the unique key of the API request, will be used in the selector too.
+ * @param { String } options.url the url for the API request.
+ * @param { Object } options.params the API get request params.
+ * @param { Object } options.payload the API payload which will be passed also to the reducer.
```

Comment:
I'm confused, didn't see the `params` before. Didn't we decide to use the `payload` as the `params`?

Reply:
Yes, sorry about that.. it was written before the discussion in #7301

Reply:
These are the API request params, payload will be passed to the reducer if needed

Reply:
I get it now, sorry for the confusion.
In this case, can the `url` and the `params` move to the `payload`?

Then we will keep in the state as **the payload that caused this response**.
The payload can be used as a hack to pass more data to the reducer but I don't think it's a valid use-case.

Reply:
currently we have a **technical debt** in many components that use the payload to pass some `id` or `controller` to set substates. soon after we'll replace all of those reducers by the main API reducer #7222 we could drop the payload from the API actions in my opinion - and we will move the `key`, `url` and `params` into the payload as you mentioned.

I think moving all of the current API actions into an action creator would really make it easy as here we only receive those as function arguments, but we can play with the structure and manage it in one place.